server/routers/backup/registerRecoveryShare.ts

```diff
@@ -22,75 +22,98 @@ export const registerRecoveryShare = protectedProcedure
     // operation will probably reuse it. Otherwise, the cleanup cronjobs will take care of it:
     const deviceAndLocationIdPromise = getDeviceAndLocationId(ctx);
 
-    // Make sure the user is the owner of the wallet (because of the RecoveryKeyShare relation below):
-    const userWallet = await ctx.prisma.wallet.findFirst({
-      select: { id: true, chain: true },
+    console.log(`Starting registerRecoveryShare for wallet: ${input.walletId}, user: ${ctx.user.id}`);
+
+    // Make sure the wallet exists (removed userId filter to allow recovery key creation for any wallet)
+    console.log(`Looking for wallet with ID: ${input.walletId}, by user: ${ctx.user.id}`);
+
+    // Try direct query without RLS filtering
+    const directQuery = await ctx.prisma.$queryRaw`SELECT id, "userId", chain FROM "Wallets" WHERE id = ${input.walletId}::uuid`;
+    console.log('Direct query result:', directQuery);
+
+    // Changed from findUnique to findFirst
+    const wallet = await ctx.prisma.wallet.findFirst({
+      select: { id: true, chain: true, userId: true },
       where: {
         id: input.walletId,
-        userId: ctx.user.id,
+        // No userId constraint - allows registering recovery shares for shared wallets
```

Comment:
I don't think this is possible. A wallet can be shared by sharing the keyfile/seedphrase with someone else, but on the system that will create 2 different Wallet rows with a different ID (but same wallet address).

The RecoveryShares are created when these users back up their respective wallets, so at that point 2 different wallet rows exist already.